lib/rust/parser/src/macros/registry.rs

```diff
@@ -0,0 +1,136 @@
+//! The macro registry that can be queried during the process of macro resolution.
+
+use crate::prelude::*;
+
+use crate::macros::definition::Definition;
+use crate::macros::literal::Literal;
+
+
+// ================
+// === Registry ===
+// ================
+
+/// The type of the tree that underlies the registry.
+pub type Tree = crate::data::tree::Tree<Literal,Definition>;
+
+/// The registry is responsible for the registration of macro definitions, and the querying of said
+/// definitions.
+#[derive(Clone,Debug,Default,PartialEq)]
+#[allow(missing_docs)]
+pub struct Registry {
+    tree : Tree
+}
+
+impl Registry {
+    /// Insert `definition` into the macro registry.
+    pub fn insert(&mut self, definition:Definition) {
+        self.tree.insert(definition.path().as_slice(),definition);
+    }
+
+    /// Get a reference to the root of the registry.
+    pub fn root(&self) -> &Tree {
+        &self.tree
+    }
+
+    /// Query the registry for a tree.
+    pub fn tree(&self, path:&[Literal]) -> Option<&Tree> {
+        self.tree.get(path)
+    }
```

Comment:
this method has non explanatory name. ITs shoudl be "get", or "subtree". The same applies to method below